source/Octopus.Tentacle.Client/Scripts/ScriptExecutionOrchestrator.cs

```diff
@@ -461,9 +461,15 @@ await asyncHalibutFeature
                             clientOperationMetricsBuilder,
                             CancellationToken.None);
 
-                    await actionTask.WaitTillCompletion(onCancellationAbandonCompleteScriptAfter, scriptExecutionCancellationToken);
+                    var actionTaskCompletionResult = await actionTask.WaitTillCompletion(onCancellationAbandonCompleteScriptAfter, scriptExecutionCancellationToken);
```

Comment:
This wasn't part of the change. But when I was trying to get failures to happen in tests, I noticed that the `catch` was never getting hit.

Looking at it closer, it's because `WaitTillCompletion` doesn't actually await the task, it just waits for it to complete in some way.

Everywhere else that uses `WaitTillCompletion` follows this pattern. So I think we need to do it here as well. Otherwise we don't log that we couldn't clean up the working directory.